tools/benchmark/src/test/CustomOptionBenchmark.test.ts

```diff
@@ -16,4 +18,17 @@ describe("`benchmarkCustom` function", () => {
 			type: BenchmarkType.OwnCorrectness,
 		});
 	});
+
+	it.only("check BenchmarkCustom can handle an error", async () => {
+		const errorMessage = "INTENTIONAL error to test error handling";
+
+		const result = benchmarkCustom({
+			title: `test`,
+			type: BenchmarkType.Measurement,
+			run: async () => {
+				throw new Error(errorMessage);
+			},
+		});
+		expect(result.err?.message).to.equal(errorMessage);
```

Comment:
Still investigating why the `err` object is `undefined`.

Reply:
It's because calling `benchmarkCustom()` will not produce the result of running the test, just like calling `it()` doesn't do it. Calling either of those just _defines_ a test that Mocha will then execute at a later time. If you look at the definition of `benchmarkCustom()`, it returns a `test` object which is basically a Mocha `it()` function. And the way things work, we emit the benchmark result (including whether something failed or not) in a "benchmark end" event on the test object.

I _think_ (but not sure) one way we could test this is by doing something like this (depends on also applying changes based on my other comment, so the runner always emits the "benchmark end" event):

```typescript
describe("validate contents of 'benchmark end' event", () => {
	const expectedErrorMessage = "INTENTIONAL error to test error handling";
	let benchmarkEndPayloadIsCorrect: boolean = false;

	const testObject = benchmarkCustom({
		title: `test`,
		type: BenchmarkType.Measurement,
		run: async () => {
			throw new Error(expectedErrorMessage);
		},
	});
	// Immediately after *defining* (not running) the test, set up a listener for the event that should be emitted
	// when the test actually runs, where we validate the event payload is correct.
	// This should run before afterEach(), where we can do normal assertions which mocha will understand if they fail
	// (vs who knows what happens if we just assert/throw from this event listener while mocha is trying to do its thing).
	testObject.on("benchmark end", (benchmarkResult: BenchmarkResult) => {
		const maybeError = benchmarkResult as unknown as BenchmarkError;
		if (maybeError.error === expectedErrorMessage){
			benchmarkEndPayloadIsCorrect	= true;
		}
	});

	// Use afterEach to validate that the test ran, it emitted "benchmark end", and the listener for that event
	// verified the event payload was correct.
	afterEach(() => {
		assert.equal(benchmarkEndPayloadIsCorrect, true);
		benchmarkEndPayloadIsCorrect = false; // Reset, so other tests (if any) start with the correct state
	});
});
```